Used language name and native name for looking in dubbed video mappings.

mrpau-richard opened the pull request:
This fixes #38 & [#5294](https://github.com/learningequality/ka-lite/issues/5294)

Used language name and native name as reference when looking in [dubbed video mappings](https://docs.google.com/spreadsheets/d/1k5xh2UXV3EchRHnYzeP6-YGKrmba22vsltGSuU9bL88/edit#gid=0).

It may also solve other languages that use the native name in [dubbed video mappings](https://docs.google.com/spreadsheets/d/1k5xh2UXV3EchRHnYzeP6-YGKrmba22vsltGSuU9bL88/edit#gid=0).

I uploaded a sample screenshot of `Swahili` when downloading videos.
![screen shot 2016-09-28 at 9 07 44 am](https://cloud.githubusercontent.com/assets/8663934/18898304/562ead5c-8563-11e6-9739-90868cbd4d2f.png)

I also uploaded the `Swahili` language pack for testing.
[sw.zip](https://github.com/fle-internal/content-pack-maker/files/496997/sw.zip)

/cc @aronasorman @jamalex @mrpau-eduard @benjaoming

mrpau-richard committed:
Look up for the lang code native name if the lang name data is null.

mrpau-eduard committed:
Add sw language pack

benjaoming commented:
@mrpau-richard and @mrpau-eduard great work

Could you put a comment about the failing tests? I don't know why they're failing of if it's related. If they are failing because of some other issue, we should get that fixed since there's no use having a repo with CI when the tests are broken :)

mrpau-richard commented:
@benjaoming I think the reason why it's failing because this [crowdin api link](https://api.crowdin.com/api/project/ka-lite/download/en.zip?key=dummy%3E) is not valid.

benjaoming commented:
We should use encrypted Travis variables and use our API key so tests can work again:

https://docs.travis-ci.com/user/environment-variables/#Encrypted-Variables

@aronasorman can you send me an email with the API key and also add me as an admin to this repo?

mrpau-eduard committed:
use swa lang code for Swahili language pack

benjaoming commented on Makefile:
```diff
@@ -12,8 +12,8 @@ es: pex
-sw: pex
-	PEX_MODULE=contentpacks ./makecontentpacks ka-lite sw 0.16 --out=out/langpacks/sw.zip --no-assessment-resources --videolang=sw
+swa: pex
+	PEX_MODULE=contentpacks ./makecontentpacks ka-lite swa 0.16 --out=out/langpacks/swa.zip --no-assessment-resources --videolang=swa --contentlang=swa
```
This means changing `all_metadata.json` on Pantry, but also the translation language code on Crownin. Why is it necessary?

The correct ISO 639-1 is `sw`.

It might be better to fix this by altering the queries for upstream Khan?

mrpau-eduard replied:
Ok @benjaoming will stick on your idea. 
As of now what I am trying to do is to build the `swa` language pack so that I can debug/compare it on the `sw` language pack.

mrpau-richard committed:
Create a function to get a related language code of the given language name.

mrpau-richard committed:
Use topic title as reference to get a list of topics.

mrpau-richard committed:
Use topic path to get a topic for videos.

mrpau-richard committed:
Build Swahili language pack for testing.

mrpau-richard committed:
Add comment on my changes.
Revert MakeFile to build all language packs.

mrpau-richard commented:
@aronasorman, @benjaoming & @cpauya can you review this PR?

It will also fix a language name that uses multiple language code.

mrpau-richard commented:
@cpauya, Yeah, I restore all languages originally supported by the makefile. Because we removed it when try to build a swahili language at build server.

cpauya commented on contentpacks/khanacademy.py:
```diff
@@ -565,9 +565,10 @@ def download_and_clean_kalite_data(url, path, lang="en") -> str:
 
 
-def retrieve_kalite_data(lang=en_lang_code, force=False, ka_domain=None, no_dubbed_videos=False) -> list:
+def retrieve_kalite_data(lang=en_lang_code, force=False, ka_domain=None, no_dubbed_videos=False, mutilple_lang_name=True) -> list:
```
I think you meant `multiple_lang_name` here?  I also don't see any use of this argument in the codes.

mrpau-richard replied:
Done removing that argument.

cpauya commented on contentpacks/utils.py:
```diff
@@ -536,6 +536,22 @@ def get_lang_native_name(lang):
+                lang_code_list.append(obj[0])
+        return lang_code_list
+    except KeyError:
+        logging.warning("No languge code found for {}. Defaulting to DEBUG.".format(lang_name))
```
Wrong spelling `languge`.

cpauya commented on contentpacks/khanacademy.py:
```diff
@@ -582,20 +583,67 @@ def retrieve_kalite_data(lang=en_lang_code, force=False, ka_domain=None, no_dubb
+    We must check if the languge code we specify has the same languge name
+        but defferent language code using get_lang_code_list function.
+    """
+    lang_list = get_lang_code_list(lang)
```
I think this is where you need to check the value of the `multiple_lang_name` switch then proceed to loop thru the language codes list only if it is `True`, else, you use the value of the `lang` switch.

mrpau-richard committed:
Fix wrong spelling string.

cpauya reviewed (approved):
I'm good with this, maybe the others can take a peek and review too.

benjaoming commented:
@mrpau-richard could you indicate if you have run the tests locally with a correct CrowdIn API key? It's relevant because since tests don't actually run on Circle, we should at least run them locally... I've added a comment about this in #10

mrpau-richard committed:
Removed extra character.

mrpau-richard commented:
@benjaoming It's still the same error in my local, when I run a test.

![screen shot 2016-10-07 at 7 28 54 am](https://cloud.githubusercontent.com/assets/8663934/19174594/7593981e-8c62-11e6-9ebe-04a2393b3da1.png)

benjaoming commented:
Yes but you're supposed to use a secret API key, I guess either @cpauya or @aronasorman has it.

mrpau-richard commented:
I already set it in my local I have a, khan and KA Lite CrowdIn Api key.

benjaoming commented:
I still don't have repo access (to merge this PR), but I think it's fine to merge -- was just looking for someone saying either "the tests are failing but I ran them locally" or "the tests are failing, but I read the output, and it's no concern". When we have these automated tests, we should keep them functional but at all costs make sure never to turn a blind eye to failures.

benjaoming commented:
> I already set it in my local I have a, khan and KA Lite CrowdIn Api key.

Your screenshot is saying `dummy>` is the string used as API key.

mrpau-richard committed:
Update variable and function name.

cpauya commented on contentpacks/khanacademy.py:
```diff
@@ -648,11 +648,11 @@ def retrieve_kalite_data(lang=en_lang_code, force=False, ka_domain=None, no_dubb
 
 
-def addin_dubbed_video_mappings(node_data, lang=en_lang_code):
+def adding_dubbed_video_mappings(node_data, lang=en_lang_code):
```
Let's use an active verb here like `add_dubbed_video_mappings`?  Here's a reference to help: http://www.powertheshell.com/bp_functionname/

mrpau-richard committed:
Rename function name.

mrpau-richard committed:
Update function name.

cpauya committed:
Refactor codes to retrieve `ka-lite` data from KA.

cpauya committed:
Move the `EN_LANG_CODE = "en"` at beginning of file for re-use.

cpauya committed:
Fix error in string formatting.

cpauya committed:
Show some logging info about languages used during build.

mrpau-richard committed:
Update README.md

mrpau-richard committed:
Change the topic filter from title to path.

mrpau-richard commented:
Hi @benjaoming I was able run a test in my local with the right error. I just deleted all `.yml` files and regenerate it. I also replace a `dummy` key with the right `KA Lite` and `KA` key.

![screen shot 2016-10-14 at 6 02 31 pm](https://cloud.githubusercontent.com/assets/8663934/19383937/611b0e76-9239-11e6-83de-6d2398c75e79.png)

mrpau-richard committed:
Rename the topic_title_name variable.
Create a get_lang_ka_name function.
Also use the ka_name as reference to find the right language name in dubbed video mappings.

cpauya commented on contentpacks/khanacademy.py:
```diff
@@ -516,147 +575,147 @@ def download_and_clean_kalite_data(url, path, lang="en") -> str:
+                    if youtube_lang == lang:
+                        youtube_ids.append(node_temp["youtube_id"])
+                        node_data.append(node_temp)
+                    elif not youtube_lang != EN_LANG_CODE:
```
To simplify, does this mean if `youtube_lang == EN_LANG_CODE`?

benjaoming replied:
I assume that it also means @mrpau-richard isn't using a PEP8 checker in his editor? :)

cpauya commented on contentpacks/khanacademy.py:
```diff
@@ -516,147 +575,147 @@ def download_and_clean_kalite_data(url, path, lang="en") -> str:
+                        youtube_ids.append(node_temp["youtube_id"])
+                        node_data.append(node_temp)
+                    elif not youtube_lang != EN_LANG_CODE:
+                        node_temp["translated_youtube_lang"] == lang
```
I assume this is supposed to be an assignment instead of an equality conditional check?

mrpau-richard committed:
Fix elif statement bug.

mrpau-richard committed:
Fix comment indention.

cpauya commented on contentpacks/khanacademy.py:
```diff
@@ -516,147 +575,147 @@ def download_and_clean_kalite_data(url, path, lang="en") -> str:
+                        Example: using pt-BR language code in the khan api will return pt translated_youtube_lang.
+                        """
+                        node_temp["translated_youtube_lang"] = lang
+                        node_data.append(node_temp)
```
Why was `youtube_ids.append(node_temp["youtube_id"])` left-out in this section?  I thought we update the `youtube_ids` list whenever we append a node?

mrpau-richard committed:
Append youtube id in the youtube id list.

mrpau-richard committed:
Fix grammar in comment.

cpauya commented on contentpacks/utils.py:
```diff
@@ -536,6 +536,51 @@ def get_lang_native_name(lang):
+def get_lang_ka_name(lang):
+    try:
+        langlookup = ujson.loads(LANGUAGELOOKUP_DATA)
+        return langlookup[lang]["native_name"]
```
Use `ka_name` here, and not `native_name`.

mrpau-richard committed:
Return empty string instead of debug string.

mrpau-richard committed:
Update languagelookup.json values.

cpauya reviewed (approved):
This looks good to me already, let's build and test a few content packs, then I can merge this.

We will address the failing tests later.

mrpau-richard committed:
Remove video attributes that cause of failing in khan api.

mrpau-eduard committed:
Use the lang_code list as reference to download the exercises files.

mrpau-eduard committed:
Fix ambiguous  logging message is for each language code.

cpauya commented:
I'm good with this, let's merge this now.
